modules/mono/godotsharp_dirs.cpp

```diff
@@ -139,18 +171,37 @@ class _GodotSharpDirs {
 #endif
 		api_assemblies_dir = api_assemblies_base_dir.path_join(GDMono::get_expected_api_build_config());
 #else // TOOLS_ENABLED
+		String platform = _get_platform_name();
 		String arch = Engine::get_singleton()->get_architecture_name();
 		String appname_safe = path::get_csharp_project_name();
-		String data_dir_root = exe_dir.path_join("data_" + appname_safe + "_" + arch);
+		String data_dir_root = exe_dir.path_join("data_" + appname_safe + "_" + platform + "_" + arch);
 		if (!DirAccess::exists(data_dir_root)) {
-			data_dir_root = exe_dir.path_join("data_Godot_" + arch);
+			data_dir_root = exe_dir.path_join("data_Godot_" + platform + "_" + arch);
+		}
+		// Fallback to directory name without the platform which was used in previous versions of Godot.
+		{
+			if (!DirAccess::exists(data_dir_root)) {
+				data_dir_root = exe_dir.path_join("data_" + appname_safe + "_" + arch);
+			}
+			if (!DirAccess::exists(data_dir_root)) {
+				data_dir_root = exe_dir.path_join("data_Godot_" + arch);
+			}
 		}
```

Comment:
Do we really need this fallback? I think this is only useful when using a newer Godot executable than the one the game was exported with, which seems pretty rare to me.

Reply:
Also this is unlikely to work anyways as the size of the managed callbacks has most likely changed since whenever such an export format was used and the export cannot be loaded anyways.